src/crimson/os/seastore/journal.cc

```diff
@@ -118,35 +119,66 @@ Journal::write_record_ret Journal::write_record(
     target);
 
   auto segment_id = current_journal_segment->get_segment_id();
+  submit_queue.emplace_back(std::make_tuple(target, current_journal_segment, std::move(to_write)));
 
-  // Start write under the current exclusive stage, but wait for it
-  // in the device_submission concurrent stage to permit multiple
-  // overlapping writes.
-  auto write_fut = current_journal_segment->write(target, to_write);
   return handle.enter(write_pipeline->device_submission
-  ).then([write_fut = std::move(write_fut)]() mutable {
-    return std::move(write_fut
-    ).handle_error(
-      write_record_ertr::pass_further{},
-      crimson::ct_error::assert_all{
-	"Invalid error in Journal::write_record"
-      }
-    );
-  }).safe_then([this, &handle] {
+  ).then([this, &handle, target, segment_id] {
     return handle.enter(write_pipeline->finalize);
-  }).safe_then([this, target, segment_id] {
+  }).then([this, target, segment_id]() -> write_record_ret {
     logger().debug(
       "write_record: commit target {}",
       target);
     if (segment_id == current_journal_segment->get_segment_id()) {
       assert(committed_to < target);
       committed_to = target;
```

Comment:
This doesn't seem to be right with these changes.

Reply:
update committed_to here to let next coming in record get right committed_to to do encode_record.

Reply:
See my reply below.